azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/cache/Cacheable.java

```diff
@@ -0,0 +1,36 @@
+package com.microsoft.azure.toolkit.lib.common.cache;
+
+import java.lang.annotation.*;
+
+@Target(value = {ElementType.METHOD})
+@Retention(value = RetentionPolicy.RUNTIME)
+@Inherited
+@Documented
+public @interface Cacheable {
+    /**
+     * jtwig expression or template for computing the cacheName dynamically.
+     * e.g. jtwig expression: {@code "$subscriptionId" }
+     * e.g. jtwig template: {@code "{{$subscriptionId}}-{{$clusterId}}"}
```

Comment:
When will we use jtwig template? And how could we pass parameters for template?

Reply:
> how could we pass parameters for template?
same as @AzureOperation,  
`$subscriptionId` means the parameter named `subscriptionId`
`@webappId` means the field named `webappId`
`@getName()` means the returned result of method `getName()`